lib/posix/pthread.c

```diff
@@ -222,15 +222,16 @@ static bool is_posix_policy_prio_valid(uint32_t priority, int policy)
 		return true;
 	}
 
-	LOG_ERR("Invalid piority %d and / or policy %d", priority, policy);
+	LOG_ERR("Invalid priority %d and / or policy %d", priority, policy);
 
 	return false;
 }
 
-static uint32_t zephyr_to_posix_priority(int32_t z_prio, int *policy)
+/* Non-static so that they can be tested in ztest */
+int zephyr_to_posix_priority(int z_prio, int *policy)
 {
 	if (z_prio < 0) {
-		__ASSERT_NO_MSG(z_prio < CONFIG_NUM_COOP_PRIORITIES);
+		__ASSERT_NO_MSG(-z_prio <= CONFIG_NUM_COOP_PRIORITIES);
```

Comment:
The change here was
```c
		__ASSERT_NO_MSG(-z_prio < CONFIG_NUM_COOP_PRIORITIES);
```
which make sure that the Zephyr priority (`z_prio`) is within `[-1, -(CONFIG_NUM_COOP_PRIORITIES - 1)]`, but Zephyr COOP priorities ranges from `[-1, -CONFIG_NUM_COOP_PRIORITIES]` instead, so change that to `<=`